src/platform/datapath_winkernel.c

```diff
@@ -2523,7 +2523,6 @@ CxPlatRecvDataReturn(
         DATAPATH_RX_IO_BLOCK* IoBlock =
             CXPLAT_CONTAINING_RECORD(Datagram, DATAPATH_RX_PACKET, Data)->IoBlock;
 
-        CXPLAT_DBG_ASSERT(Binding == NULL || Binding == IoBlock->Binding);
```

Comment:
Why?

Reply:
When a connection binds more than one bindings (i.e. after adding a new local address on the client side), we cannot assure that the Datagrams in RecvDataChain come from the same binding.